### PR TITLE
:bug: Removes okta css from affecting single-app css

### LINKED ISSUE #126: Widget CSS conflicts with my app
I like that Okta provides this sign-in widget to help me get going, but the use of reset styles and other generic styles makes the provided CSS unusable

https://github.com/okta/okta-signin-widget/blob/okta-signin-widget-1.7.0/assets/sass/common/enduser/_reset.scss
https://github.com/okta/okta-signin-widget/blob/okta-signin-widget-1.7.0/assets/sass/widgets/_mega-drop-down.scss

Specifically, the global `input` styles and the generic `.dropdown` styles conflict with styles already built in my application, forcing me to completely style the Okta widget on my own. Something like this always needs to be scoped under an Okta class so as to not collide with everything else in your clients' custom apps.

### PR DESCRIPTION
**Pull request by alanzhou-okta**
Adds a ScssContainer so css only affects that container
Moves scss imports so they only affect the ScssContainer

Resolves #126

### CODE REVIEW

File: assets/sass/okta-sign-in.scss [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
-@import 'modules/header';
-@import 'modules/beacon';
-@import 'modules/factors-dropdown';
-@import 'modules/footer';
-@import 'modules/enroll';
-@import 'modules/app-login-banner';
-@import 'modules/okta-footer';
-@import 'ie';
+
+.auth-container-container {
```
Comment (rchild-okta) [inline assets/sass/okta-sign-in.scss]: Oh nice, this works.

File: src/views/shared/ScssContainer.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+define([
+    'okta'
+  ],
+  function (Okta) {
+    return Okta.View.extend({
+      className: 'auth-container-container',
```
Comment (rchild-okta) [inline src/views/shared/ScssContainer.js]: Do we need to create a new top level div? Ideally we would use our current #okta-sign-in one (which is already supposed to be the root div that we apply all styles to).

Comment (alanzhou-okta) [inline src/views/shared/ScssContainer.js]: So when we do nested scss, it only applies the css to the children elements of the div. So we have to do it to one layer above what we want.

Comment (rchild-okta) [inline src/views/shared/ScssContainer.js]: Do these styles apply to the original parent div?

Comment (alanzhou-okta) [inline src/views/shared/ScssContainer.js]: no, if we just nest it in #okta-sign-in the styles that we apply to that div don't apply.

**Comment by alanzhou-okta** [comment]
Demo: with our current scss 
screen shot 2016-11-02 at 2 12 34 pm 2
As you can see, the other div on the page gets completely overriden by ours.

Updated: 
screen shot 2016-11-02 at 2 04 33 pm 2
The reset styles are only applied within the auth-container, means their css is not overridden.

File: assets/sass/_container.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
+.auth-container {
```
Comment (rchild-okta) [inline assets/sass/_container.scss]: We should scope this to #okta-sign-in, i.e.:
```css
#okta-sign-in.auth-container {
}
```

As well as the other places that we have this. This is to answer the question of "what happens if they have their own .auth-container styles?"

File: assets/sass/_container.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  @include max-width-mq($container-width) {
+    width: 100%;
+  }
+
+  @include max-height-mq(750px) {
+    margin-top: 0;
+  }
+}
+
+.no-beacon .auth-content {
```
Comment (rchild-okta) [inline assets/sass/_container.scss]: Does this live on the same level as auth-container? If so, we should also scope this with #okta-sign-in. If not, let's add it inside the body of the .auth-container class.

Actually in general, I am wondering why we need to set the styles to .auth-container - what happens if we just use #okta-signin-in instead?

File: assets/sass/common/enduser/_reset.scss [hunk 5]
```diff
@@ -1,56 +1,59 @@
...
+    margin: 0;
+    padding: 0;
+    border: 0;
+    outline: 0;
+    font-size: 100%;
+    font: inherit;
+    vertical-align: baseline;
+    background: transparent;
+  }
+  body {
```
Comment (rchild-okta) [inline assets/sass/common/enduser/_reset.scss]: So we shouldn't ever be in a position where body is underneath .auth-container - does this change how the widget looks if the line-height is not reset to 1? What happens if the outer page sets body line-height to something else? On our pages that we have multiple lines, does it look funny?

I'm guessing we should probably be putting line-height: 1 in the root element.

File: assets/sass/okta-sign-in.scss [hunk 6]
```diff
@@ -1,32 +1,42 @@
 @import 'compass/css3';
 
 @import 'widgets/jquery.qtip';
```
Comment (rchild-okta) [inline assets/sass/okta-sign-in.scss]: Where does this come from? And does it drop styles into the page or is it just a helper file?

File: assets/sass/okta-sign-in.scss [hunk 7]
```diff
@@ -1,32 +1,42 @@
...
-@import 'widgets/mega-drop-down';
-@import 'widgets/chosen';
 
 @import 'common/enduser/reset';
```
Comment (rchild-okta) [inline assets/sass/okta-sign-in.scss]: Does this file reset styles as well?

File: assets/sass/_container.scss (hunk 4 above)
Comment (alanzhou-okta) [inline assets/sass/_container.scss]: Sass linter complains that we can't use id's

Comment (alanzhou-okta) [inline assets/sass/_container.scss]: So I haven't tried using a combination of id & class, but only id's will cause the linter to complain

File: assets/sass/common/enduser/_reset.scss (hunk 5 above)
Comment (alanzhou-okta) [inline assets/sass/common/enduser/_reset.scss]: What if we just delete it? I don't see a way to really keep the body css without having some kind of conflict with the outer page.

File: assets/sass/_container.scss (hunk 4 above)
Comment (alanzhou-okta) [inline assets/sass/_container.scss]: Re: same level. Yes, no-beacon is another class of #okta-sign-in

File: assets/sass/okta-sign-in.scss (hunk 7 above)
Comment (alanzhou-okta) [inline assets/sass/okta-sign-in.scss]: This is the only reset style page

File: assets/sass/_container.scss (hunk 4 above)
Comment (rchild-okta) [inline assets/sass/_container.scss]: In this case, we know we want to use an id - let's just disable that lint warning in this file and the other one where we'll use the id's

File: assets/sass/common/enduser/_reset.scss (hunk 5 above)
Comment (rchild-okta) [inline assets/sass/common/enduser/_reset.scss]: We need to remove it, but we still need the style applied somewhere - we should put it on .auth-container.

You should also update the sample you're running to apply a different line-height and verify that w/out our update, it looks different (and with it, it looks like it should)

File: assets/sass/okta-sign-in.scss (hunk 7 above)
Comment (rchild-okta) [inline assets/sass/okta-sign-in.scss]: Oh right, this is the one we're modifying

File: assets/sass/common/enduser/_reset.scss (hunk 5 above)
Comment (alanzhou-okta) [inline assets/sass/common/enduser/_reset.scss]: I moved all the stuff that would apply to the root elements (html and body) to _container.scss including some stuff from _base.scss. Line-height was overriden by line-height from _base.scss, but _container.scss should now contain all root scss.

File: .scss-lint.yml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 scss_files: "assets/sass/**/*.scss"
 plugin_directories: ['.scss-linters']
-exclude: ["assets/sass/widgets/*.scss", "assets/sass/common/**/*.scss"]
+exclude: ["assets/sass/widgets/*.scss", "assets/sass/common/**/*.scss", "assets/sass/_container.scss", "assets/sass/okta-sign-in.scss"]
```
Comment (rchild-okta) [inline .scss-lint.yml]: We shouldn't ignore the whole file - just the line itself, i.e. http://stackoverflow.com/questions/32833542/having-scss-lint-ignore-a-particular-line

File: assets/sass/_base.scss [hunk 9]
```diff
@@ -1,20 +1,3 @@
-html {
```
Comment (rchild-okta) [inline assets/sass/_base.scss]: Can you try importing this into okta-core and making sure that this does not change the styling of the login page?

File: assets/sass/_container.scss [hunk 10]
```diff
@@ -0,0 +1,48 @@
+#okta-sign-in.auth-container {
+
+  // Html and body styles in _base.scss that were moved here
+  height: 100%;
+  font-family: $fonts;
+  font-size: $font-size;
+  font-weight: $font-weight-body;
+  line-height: 1.4;
+  background-color: $secondary-bg-color;
+  min-width: 100%;
```
Comment (rchild-okta) [inline assets/sass/_container.scss]: We should remove the properties in this section that are being overridden below.

File: assets/sass/_container.scss [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  min-height: 100%;
+  color: $medium-text-color;
+  position: relative;
```
Comment (rchild-okta) [inline assets/sass/_container.scss]: Also I am wondering if this affects anything - if it doesn't change the appearance, we should remove it.

File: assets/sass/_container.scss [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  overflow: auto;
```
Comment (rchild-okta) [inline assets/sass/_container.scss]: We probably don't need this rule.

**Comment by alanzhou-okta** [comment]
When we modularize our okta code to #okta-sign-in.auth-container, this makes the selector very specific. However, in okta-theme, we have okta specific selectors for #okta-sign-in, and these selectors are not overriding the generic selectors in #okta-sign-in.auth-container. 
There are many scss files that are being used outside the auth-container, and that means we have to import them to okta-core. The problem is there are some that currently may not be being used, but may be used in the future- do we import them or not?
A lot of double imports going on- importing almost exactly the same file in okta-core and okta-sign-in, except the okta-sign-in file has some selectors that are not applicable to the auth-container i.e. html, body.

**Pull request closed without merging**